spec/controllers/passengers_controller_spec.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe PassengersController do
+
+  let!(:admin) { create :user, :admin }
+  before :each do
+    session[:user_id] = admin.id
+  end
+
+  describe 'POST set_status' do
+    context 'with broken mailer' do
+      context 'changing active to archived' do
+        it 'saves the passenger and displays a warning message' do
+          allow(PassengerMailer).to receive(:notify_archived).and_raise Net::SMTPFatalError
+
+          @passenger = create :passenger, :permanent, active_status: 'active'
+          post :set_status, params: { id: @passenger.id, status: 'archived' }
+          @passenger.reload
+
+          expect(@passenger.active_status).to eq 'archived'
+          expect(response).to redirect_to(passengers_path)
+          expect(flash[:warning]).to match(/the passenger was unable to be notified of their status change via email/)
+        end
+      end
+      context 'changing archived to active' do
+        it 'saves the passenger and displays a warning message' do
+          allow(PassengerMailer).to receive(:notify_active).and_raise Net::SMTPFatalError
+
+          @passenger = create :passenger, :permanent, active_status: 'archived'
+          post :set_status, params: { id: @passenger.id, status: 'active' }
+          @passenger.reload
+
+          expect(@passenger.active_status).to eq 'active'
+          expect(response).to redirect_to(passengers_path)
+          expect(flash[:warning]).to match(/the passenger was unable to be notified of their status change via email/)
+        end
+      end
+    end
+  end
+
+  describe 'POST create' do
+    context 'with broken mailer' do
+      context 'creating a pending passenger' do
+        it 'saves the passenger and displays a warning message' do
+          allow(PassengerMailer).to receive(:notify_pending).and_raise Net::SMTPFatalError
+
+          new_passenger_params = {
+            passenger: {
```

Comment:
You can do `attributes_for :passenger, :permanent` to make Factory Bot build this hash.